[ARM/CI] Enable manual CI triggering for arm and armel

hqueue opened the pull request:
1. Enable following manual CI triggering for `arm` and `armel`.

    innerloop Ubuntu arm Release Cross Build
    innerloop Ubuntu arm Debug Cross Build
    innerloop Ubuntu16.04 arm Release Cross Build
    innerloop Ubuntu16.04 arm Debug Cross Build
    innerloop Tizen armel Release Cross Build
    innerloop Tizen armel Debug Cross Build

    Which replaces following old style commands.
      innerloop linuxarmemulator HardFP Release
      innerloop linuxarmemulator HardFP Debug
      innerloop linuxarmemulator SoftFP Release
      innerloop linuxarmemulator SoftFP Debug

2. Prepare `bin/build.tar.gz` correctly to run unit tests in coreclr CI for `arm` and `armel`. Before netci.groovy made old-fashioned `build.tar.gz`.

3. Add some options to arm32 CI script.

4. Rename scripts/arm32_ci_script.sh to cross/arm32_ci_script.sh

Related issues: https://github.com/dotnet/core-setup/issues/790

hqueue commented:
@dotnet-bot help

hqueue commented:
@dotnet-bot test ci please
@dotnet-bot test innerloop linuxarmemulator HardFP Release
@dotnet-bot test innerloop linuxarmemulator SoftFP Release

hqueue commented:
@dotnet-bot test ci please
@dotnet-bot test innerloop linuxarmemulator HardFP Release
@dotnet-bot test innerloop linuxarmemulator SoftFP Release

hqueue commented:
@dotnet-bot test ci please
@dotnet-bot test innerloop linuxarmemulator HardFP Release
@dotnet-bot test innerloop linuxarmemulator SoftFP Release

hqueue commented:
@dotnet-bot test ci please
@dotnet-bot test innerloop linuxarmemulator HardFP Release
@dotnet-bot test innerloop linuxarmemulator SoftFP Release

hqueue commented:
@dotnet-bot test ci please
@dotnet-bot test innerloop linuxarmemulator HardFP Release
@dotnet-bot test innerloop linuxarmemulator SoftFP Release

hqueue commented:
@dotnet-bot test ci please
@dotnet-bot test innerloop linuxarmemulator HardFP Release
@dotnet-bot test innerloop linuxarmemulator SoftFP Release

hqueue commented:
@dotnet-bot test ci please
@dotnet-bot test innerloop linuxarmemulator HardFP Release
@dotnet-bot test innerloop linuxarmemulator SoftFP Release

hqueue commented:
@dotnet-bot test ci please
@dotnet-bot test innerloop linuxarmemulator HardFP Release
@dotnet-bot test innerloop linuxarmemulator SoftFP Release

hqueue commented:
@dotnet-bot test ci please
@dotnet-bot test innerloop linuxarmemulator HardFP Release
@dotnet-bot test innerloop linuxarmemulator SoftFP Release

hqueue commented:
@gkhanna79 @mellinoe PTAL.
I've update `netci.groovy` and CI script.  It is also required to enable unit tests in coreclr CI for arm and armel.

hqueue commented:
@dotnet-bot test ci please
@dotnet-bot test innerloop linuxarmemulator HardFP Release
@dotnet-bot test innerloop linuxarmemulator SoftFP Release
@dotnet-bot test innerloop linuxarmemulator HardFP Debug
@dotnet-bot test innerloop linuxarmemulator SoftFP Debug

mellinoe commented:
This looks good to me. @gkhanna79 do you want to take a quick look?

weshaggard commented on scripts/arm32_ci_script.sh:
```diff
@@ -9,23 +9,23 @@ function usage {
     echo '    corefx source is at ~/cfx'
```
Could you also please move this script under the cross folder like the other similar scripts?

hqueue replied:
I will rename it.

hqueue commented:
@dotnet-bot test ci please
@dotnet-bot test innerloop linuxarmemulator HardFP Release
@dotnet-bot test innerloop linuxarmemulator SoftFP Release
@dotnet-bot test innerloop linuxarmemulator HardFP Debug
@dotnet-bot test innerloop linuxarmemulator SoftFP Debug

hqueue commented:
@weshaggard I've renamed the script scripts/arm32_ci_script.sh to cross/arm32_ci_script.sh. Is it ok? 

@mellinoe @gkhanna79 @weshaggard PTAL
After @mellinoe  approved, I've renamed the script following @weshaggard feedback.
As a result, four arm32 related CI is failed due to the script is not found, because netci.groovy is not applied yet. I think it will work without problem after merging this PR.
At least  it work well before renaming the script.
Therefore please merge this PR :)

If there is more change required, I think we can apply it after merging this PR.

weshaggard commented:
>@weshaggard I've renamed the script scripts/arm32_ci_script.sh to cross/arm32_ci_script.sh. Is it ok?

Yes that is fine with me.

weshaggard commented on cross/arm32_ci_script.sh:
```diff
@@ -9,23 +9,23 @@ function usage {
     echo '    corefx source is at ~/cfx'
     echo '$ cd ~/cfx'
     echo '$ ./scripts/arm32_ci_script.sh'
```
nit: as probably should update this comment but you can do that after this PR is merged.

hqueue replied:
Sure, I will update and revise the script. Thanks.
